torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -2026,6 +2022,25 @@ void ONNXShapeTypeInference(
   }
 
   SpecialPostProcess(n);
+  // Get data propagation result from ONNX shape inference
+  for (const auto& output: n->outputs()) {
+    const auto inferred_shape_pair = inferred_shape_data.find(torch_to_onnx_output_name[output->debugName()]);
+    if (inferred_shape_pair != inferred_shape_data.end()) {
+      const auto& inferred_shape = inferred_shape_pair->second;
+      int rank = inferred_shape.dim_size();
+      std::vector<::c10::ShapeSymbol> final_shape(rank);
+      for (int i = 0; i < rank; ++i) {
+        final_shape[i] = ONNXDimToShapeSymbol(inferred_shape.dim(i), symbol_dim_map);
+      }
+      c10::SymbolicShape shape_value(final_shape);
+      // Store data propagation result into shapeValueMap
+      ConstantValueMap::SetShapeValue(output->debugName(), shape_value);
+      // Use original name in PyTorch graph instead of temporary name in intermediate ONNX graph
+      inferred_shape_data[output->debugName()] = inferred_shape;
```

Comment:
My concern here is it seems (very) possible `inferred_shape_data` after a few calls will be in a state of using both names in "PyTorch" graph, and names in intermediate ONNX graph. Since this part is only mapping `n->outputs()` back to "pytorch name". The changes of mapping from "pytorch name" to ONNX name at the beginning of `ONNXShapeTypeInference` was not reverted.

Reply:
Thanks for catching it! Yes your concern is valid and now it will only add node's output into inferred_shape_data for furtuer data propagation.